### ISSUE TITLE
Question regarding Shipment Pickup and Drop

### ISSUE THREAD

**Issue by noob-slayer** [VROOM-Project/vroom#420]
Let's say I have 20 locations to visit with 2 vehicles (Assume 10 locations per vehicle). I have to drop and pick up shipments at each location, but the condition is "All the shipment drops have to be made in the first half of the day, and pickups in the second half of day. 

Ex: Consider my route as:  Depot - A - B - C - D - Depot

The vehicle will take all shipments from the Depot and DROP them at locations A, B, C, D in the first half of the day (by 12 PM).  The vehicle has to stop at D (the last location in the drop route). At D, the vehicle will have no shipment. After 12 PM, the vehicle has to start PICKING up shipments from locations D, C, B, A, and back to the depot. 

So basically the route has to be Depot - A - B - C - D - C - B - A - Depot. 

And there is also a vehicle capacity constraint on both the Drop route (Depot to D) and Pickup route (D to Depot)

Is this possible?

**Comment by jcoupey** [comment]
> All the shipment drops have to be made in the first half of the day, and pickups in the second half of day

Sounds like a timing constraint. Those are set using `time_windows` keys for tasks.

**Comment by noob-slayer** [comment]
> The vehicle will take all shipments from the Depot and DROP them at locations A, B, C, D in the first half of the day (by 12 PM). The vehicle has to stop at D (the last location in the drop route). At D, the vehicle will have no shipment. After 12 PM, the vehicle has to start PICKING up shipments from locations D, C, B, A, and back to the depot.
> 
> So basically the route has to be Depot - A - B - C - D - C - B - A - Depot.

But how to handle this scenario? Returning back on the same route?

**Comment by jcoupey** [comment]
What about using one job per step:

Depot -> vehicle start
A -> job with `delivery` key and "morning" time window
B -> job with `delivery` key and "morning" time window
C -> job with `delivery` key and "morning" time window
D -> job with `delivery` key and "morning" time window
D -> job with `pickup` key and "afternoon" time window
C -> job with `pickup` key and "afternoon" time window
B -> job with `pickup` key and "afternoon" time window
A -> job with `pickup` key and "afternoon" time window
Depot -> vehicle end

**Comment by jcoupey** [comment]
Looks like this can be closed as answered.
